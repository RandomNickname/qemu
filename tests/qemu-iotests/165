#! /bin/bash
#
# qemu-img dd test for conv option
#
# Copyright (C) 2016 Reda Sallahi
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

owner=fullmanet@gmail.com

seq="$(basename $0)"
echo "QA output created by $seq"

here="$PWD"
status=1

_cleanup()
{
    _cleanup_test_img
    rm -f "$TEST_IMG.out" "$TEST_IMG.out.dd"
}
trap "_cleanup; exit \$status" 0 1 2 3 15

. ./common.rc
. ./common.filter
. ./common.pattern

_supported_fmt raw
_supported_proto file
_supported_os Linux

echo
echo "== Creating image =="

size=1M
_make_test_img $size
_check_test_img

$QEMU_IO -c "write -P 0xa 215k 212k" "$TEST_IMG" | _filter_qemu_io

echo
echo "== Converting the image with dd with conv=excl =="

$QEMU_IMG dd if="$TEST_IMG" of="$TEST_IMG.out" conv=excl,notrunc \
          -O "$IMGFMT" status=none

TEST_IMG="$TEST_IMG.out" _check_test_img

dd if="$TEST_IMG" of="$TEST_IMG.out.dd" conv=excl status=none

echo
echo "== Compare the images with qemu-img compare =="

$QEMU_IMG compare "$TEST_IMG.out.dd" "$TEST_IMG.out"

echo
echo "== Converting the image with dd with conv=excl =="

$QEMU_IMG dd if="$TEST_IMG" of="$TEST_IMG.out" conv=excl,notrunc status=none \
          -O "$IMGFMT" 2>&1 | sed -e "s#$TEST_DIR#TEST_DIR#g" \
                                  -e "s#$IMGFMT#IMGFMT#g"

echo
echo "== Creating image =="

_make_test_img $size
_check_test_img

$QEMU_IO -c "write -P 0xa 481k 329k" "$TEST_IMG" | _filter_qemu_io

echo
echo "== Converting the image with dd with conv=nocreat =="

$QEMU_IMG dd if="$TEST_IMG" of="$TEST_IMG.out" conv=nocreat,notrunc \
          -O "$IMGFMT" status=none

TEST_IMG="$TEST_IMG.out" _check_test_img

dd if="$TEST_IMG" of="$TEST_IMG.out.dd" conv=nocreat status=none

echo
echo "== Compare the images with qemu-img compare =="

$QEMU_IMG compare "$TEST_IMG.out.dd" "$TEST_IMG.out"

rm -f "$TEST_IMG.out" "$TEST_IMG.out.dd"

echo
echo "== Converting the image with dd with conv=nocreat =="

$QEMU_IMG dd if="$TEST_IMG" of="$TEST_IMG.out" conv=nocreat,notrunc \
          -O "$IMGFMT" 2>&1 status=none | sed -e "s#$TEST_DIR#TEST_DIR#g" \
                                              -e "s#$IMGFMT#IMGFMT#g"

echo
echo "*** done"
rm -f "$seq.full"
status=0
