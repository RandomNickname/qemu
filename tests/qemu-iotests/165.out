QA output created by 165

== Creating image ==
Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=1048576
No errors were found on the image.
wrote 217088/217088 bytes at offset 220160
212 KiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

== Converting the image with dd with conv=excl ==
No errors were found on the image.

== Compare the images with qemu-img compare ==
Images are identical.

== Converting the image with dd with conv=excl ==
qemu-img: failed to open 'TEST_DIR/t.IMGFMT.out': File exists

== Creating image ==
Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=1048576
No errors were found on the image.
wrote 336896/336896 bytes at offset 492544
329 KiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

== Converting the image with dd with conv=nocreat ==
No errors were found on the image.

== Compare the images with qemu-img compare ==
Images are identical.

== Converting the image with dd with conv=nocreat ==
qemu-img: Failed to open 'TEST_DIR/t.IMGFMT.out': No such file or directory

*** done
